torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -2077,47 +2068,55 @@ void decomposeLinearOps(Block* block) {
 // Replace 'operation' with 'operation_copy' to guard alias operations.
 // Supports View, Reshape, Squeeze, and Unsqueeze
 void replaceAliasOpsWithCopy(std::shared_ptr<Graph>& graph, Block* block) {
-  static std::unordered_map<Symbol, Symbol> op_mapping(
+  static std::unordered_map<Symbol, Symbol> alias_to_copy_mapping(
       {{aten::view, prim::view_copy},
        {aten::reshape, prim::reshape_copy},
        {aten::squeeze, prim::squeeze_copy},
        {aten::unsqueeze, prim::unsqueeze_copy}});
 
-  std::vector<Node*> maybe_alias_nodes;
+  std::vector<Node*> maybe_safe_alias_nodes;
   for (Node* n : block->nodes()) {
     for (Block* b : n->blocks()) {
       replaceAliasOpsWithCopy(graph, b);
     }
-    if (op_mapping.find(n->kind()) != op_mapping.end()) {
-      maybe_alias_nodes.push_back(n);
+    if (alias_to_copy_mapping.find(n->kind()) != alias_to_copy_mapping.end()) {
+      maybe_safe_alias_nodes.push_back(n);
     }
   }
 
-  auto alias_db = std::make_unique<AliasDb>(graph);
-  for (Node* n : maybe_alias_nodes) {
-    if (!alias_db->safeToChangeAliasingRelationship(
-            n->input(0), n->output(0))) {
-      continue;
-    }
+  auto alias_db = std::make_shared<AliasDb>(graph);
```

Comment:
nitpick, don't see we sharing alias_db around, any reason that we changed it from unique_ptr to shared_ptr?